src/components/Personalization/createViewChangeHandler.js

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import isNonEmptyArray from "../../utils/isNonEmptyArray";
+
+export default ({ getView, executeViewDecisions, collect }) => {
+  return ({ viewName }) => {
+    const viewDecisions = getView(viewName);
+
+    if (isNonEmptyArray(viewDecisions)) {
+      executeViewDecisions(viewDecisions);
+      return;
+    }
+
+    const xdm = { web: { webPageDetails: { viewName } } };
+
+    collect({ meta: {}, xdm });
```

Comment:
Why send a display event when no decisions were rendered?

Reply:
We send a notification of a view "displayed" even if we didn't have anything in the cache to render for that specific view. This is needed for Target reporting.